model/types.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package model
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+type WebhookHeader struct {
+	Key          string  `json:"key"`
+	Value        *string `json:"value,omitempty"`
+	ValueFromEnv *string `json:"value_from_env,omitempty"`
+}
+
+type Headers []WebhookHeader
+
+func (wh Headers) Value() (driver.Value, error) {
+	return json.Marshal(wh)
+}
+
+func (wh *Headers) Scan(databaseValue interface{}) error {
```

Comment:
`databaseValue` can also be `nil` here. `panic`

Reply:
This will fall down to the `default` case.

Reply:
Right, but with error. If we do not have headers, we want empty list, right? The whole DB scan will return error here.

Reply:
Oh yeah, you're right. I added at some point a test case for that scenario and since test were passing I thought the database driver took care of that (just using `nil` from `NULL`). Adding the test case (which is mising, obviously) and handling nulls explicitly. Good catch.